speechbrain/utils/distributed.py

```diff
@@ -10,12 +10,55 @@
 import datetime
 import os
 from functools import wraps
+from typing import Optional
 
 import torch
 
 MAIN_PROC_ONLY: int = 0
 
 
+def rank_prefixed_message(message: str, rank: Optional[int]) -> str:
+    r"""Prefix a message with the rank of the process.
+
+    Arguments
+    ---------
+    message : str
+        The message to prefix.
+    rank : int or None
+        The rank of the process, or None if the rank is unknown.
+
+    Returns
+    -------
+    str
+        The message prefixed with the rank, if known.
+    """
+    if rank is not None:
+        return f"[rank: {rank}] {message}"
+    return message
+
+
+def get_rank() -> Optional[int]:
+    r"""Get the rank of the current process.
+
+    This code is taken from the Pytorch Lightning library:
+    https://github.com/Lightning-AI/pytorch-lightning/
```

Comment:
Perhaps a permalink to the actual file where it was found could be helpful here

Reply:
Done.